core/shared/src/main/scala/matryoshka/Based.scala

```diff
@@ -27,3 +27,6 @@ import simulacrum.typeclass
 @typeclass trait Based[T] {
   type Base[A]
 }
+
+@java.lang.SuppressWarnings(scala.Array("org.wartremover.warts.PublicInference"))
+object Based
```

Comment:
WartRemover has bugs with the PublicInference wart. We should probably actually suppress it globally.

Reply:
@djspiewak It was suppressed globally but this is happening (as far as I can tell) specifically because Simulacrum is adding methods to the companion `object` that are not specifying the return type.